R/io-ipc-functions.R

```diff
@@ -0,0 +1,145 @@
+#' Lazily read from an Arrow IPC (Feather v2) file or multiple files via glob
+#' patterns
+#'
+#' This allows the query optimizer to push down predicates and projections to
+#' the scan level, thereby potentially reducing memory overhead.
+#'
+#' @inheritParams rlang::args_dots_empty
+#' @inheritParams pl_scan_parquet
+#' @param columns Columns to select. Accepts a vector of column indices
+#' (starting at zero) or a vector of column names.
+#' @param memory_map A logical. If `TRUE`, try to memory map the file. This can
+#' greatly improve performance on repeated queries as the OS may cache pages.
+#' Only uncompressed Arrow IPC files can be memory mapped.
+#' @param hive_partitioning Infer statistics and schema from Hive partitioned
+#' sources and use them to prune reads. If `NULL` (default), it is automatically
+#' enabled when a single directory is passed, and otherwise disabled.
+#' @param hive_schema A list containing the column names and data types of the
+#' columns by which the data is partitioned, e.g.
+#' `list(a = pl$String, b = pl$Float32)`. If `NULL` (default), the schema of
+#' the Hive partitions is inferred.
+#' @param try_parse_hive_dates Whether to try parsing hive values as date /
+#' datetime types.
+#' @param include_file_paths Character value indicating the column name that
+#' will include the path of the source file(s).
+#'
+#' @details
+#' If `memory_map` is set, the bytes on disk are mapped 1:1 to memory. That
+#' means that you cannot write to the same filename, e.g.
+#' `pl$read_ipc("my_file.arrow")$write_ipc("my_file.arrow")` will fail.
+#'
+#' @rdname IO_scan_ipc
+#' @examplesIf requireNamespace("arrow", quietly = TRUE) && arrow::arrow_with_dataset()
+#' temp_dir <- tempfile()
+#' # Write a hive-style partitioned arrow file dataset
+#' arrow::write_dataset(
+#'   mtcars,
+#'   temp_dir,
+#'   partitioning = c("cyl", "gear"),
+#'   format = "arrow",
+#'   hive_style = TRUE
+#' )
+#' list.files(temp_dir, recursive = TRUE)
+#'
+#' # If the path is a folder, Polars automatically tries to detect partitions
+#' # and includes them in the output
+#' pl$scan_ipc(temp_dir)$collect()
+#'
+#' # We can also impose a schema to the partition
+#' pl$scan_ipc(temp_dir, hive_schema = list(cyl = pl$String, gear = pl$Int32))$collect()
+pl__scan_ipc <- function(
+    source,
+    ...,
+    columns = NULL,
```

Comment:
Is this needed?

Reply:
This appears in read_ipc but not scan_ipc. IMO it's not super useful since it's a simple select() call at the end. No problem to remove if you want.